projects/plugins/jetpack/extensions/shared/register-jetpack-block.js

```diff
@@ -57,3 +58,24 @@ export default function registerJetpackBlock( name, settings, childBlocks = [],
 
 	return result;
 }
+
+/**
+ * Wrapper around registerJetpackBlock to register a block by specifying its metadata.
+ *
+ * @param {object }metadata - Metadata of the block (content of block.json)
+ * @param {object} settings - See registerJetpackBlock.
+ * @param {object} childBlocks - See registerJetpackBlock.
+ * @param {boolean} prefix - See registerJetpackBlock.
+ * @returns {object|boolean} Either false if the block is not available, or the results of `registerBlockType`
+ */
+export function registerJetpackBlockFromMetadata( metadata, settings, childBlocks, prefix ) {
```

Comment:
It's interesting, I would have expected a modified `registerJetpackBlock`, ready to handle different type of data passed to it, but I suppose a new function makes sense as well. 👍 It will probably be easier to understand that way!